### PR TITLE
[processing] Fix running commands on layers with open options or credential options

### PR DESCRIPTION
**Pull request by nyalldawson**
(temporarily includes https://github.com/qgis/QGIS/pull/57868)

These improvements fix generation of GDAL commands when the input layers have open options specified (translating these to -oo arguments wherever the GDAL command supports these), or when credential options are specified for the QGIS layer (translating to --config arguments).

In addition to other fixes, this permits the GDAL tools to be run directly on layers loaded from eg a S3 bucket via GDAL vsis3.

### CODE REVIEW

File: python/plugins/processing/algs/gdal/GridAverage.py [hunk 1]
```diff
@@ -180,6 +182,9 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-of')
         arguments.append(output_format)
 
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/GridAverage.py:185]: Support for open options in this tool was added in GDAL 3.7. Maybe we should check GDAL version here to support old platforms?

File: python/plugins/processing/algs/gdal/GridDataMetrics.py [hunk 2]
```diff
@@ -194,6 +196,9 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-of')
         arguments.append(output_format)
 
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/GridDataMetrics.py:199]: Same as above

File: python/plugins/processing/algs/gdal/GridInverseDistance.py [hunk 3]
```diff
@@ -203,6 +205,9 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-of')
         arguments.append(output_format)
 
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/GridInverseDistance.py:208]: Same as above

File: python/plugins/processing/algs/gdal/GridInverseDistanceNearestNeighbor.py [hunk 4]
```diff
@@ -188,6 +190,9 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-of')
         arguments.append(output_format)
 
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/GridInverseDistanceNearestNeighbor.py:193]: Also here

File: python/plugins/processing/algs/gdal/GridLinear.py [hunk 5]
```diff
@@ -160,6 +162,9 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-of')
         arguments.append(output_format)
 
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/GridLinear.py:165]: And here

File: python/plugins/processing/algs/gdal/GridNearestNeighbor.py [hunk 6]
```diff
@@ -174,6 +174,9 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-of')
         arguments.append(output_format)
 
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/GridNearestNeighbor.py:177]: Same as above

File: python/plugins/processing/algs/gdal/rasterize.py [hunk 7]
```diff
@@ -237,6 +235,12 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-of')
         arguments.append(output_format)
 
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/rasterize.py:238]: Support for open options was added in GDAL 3.7, maybe add a check for GDAL version here?

File: python/plugins/processing/algs/gdal/rasterize_over.py [hunk 8]
```diff
@@ -119,8 +119,14 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
             extra = self.parameterAsString(parameters, self.EXTRA, context)
             arguments.append(extra)
 
-        arguments.append(ogrLayer)
-        arguments.append(inLayer.source())
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/rasterize_over.py:122]: Support for open options was added in GDAL 3.7, maybe check GDAL version here?

File: python/plugins/processing/algs/gdal/rasterize_over_fixed_value.py [hunk 9]
```diff
@@ -96,29 +93,37 @@ def commandName(self):
...
-            layerName,
+            input_details.layer_name,
             '-burn',
             str(self.parameterAsDouble(parameters, self.BURN, context)),
         ]
 
         if self.parameterAsBool(parameters, self.ADD, context):
             arguments.append('-add')
 
+        if input_details.open_options:
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/rasterize_over_fixed_value.py:115]: Same as above

**Review by alexbruy — APPROVED** [review]
Looks good to me. Just not sure if we need a check for GDAL version if a few algorithms where support for open options was added in GDAL 3.7.

File: python/plugins/processing/algs/gdal/rasterize_over_fixed_value.py (hunk 9 above)
Comment (nyalldawson) [reply python/plugins/processing/algs/gdal/rasterize_over_fixed_value.py:115]: Thanks, done

**Comment by agiudiceandrea** [comment]
Hi @nyalldawson, please have a look at the regression reported at https://github.com/qgis/QGIS/issues/61144.

**Pull request merged**